Je teste des trucs debiles

FrancoisDuchene opened the pull request:
# Easter Eggs
J'ai ajouté quelques easter eggs
- Quand on appuie sur le bouton EPL, y a des avions en papier qui apparaissent. Si on appuie plusieurs fois de suite, il y en a deux qui apparaissent de deux cotés à la fois.
- Quand on ouvre un lien youtube via le lien vers sa chaine ou via le bouton covid, des sons sont lancés.
- Quand on appuie longuement sur le bouton comu, (spoiler) une photo de chat triste et une musique triste s'affichent/sont jouées pendant 60 secondes. Parce qu'un autre son est joué pendant la lecture de la musique, le mode de lecture est changé en "plusieurs sons en même temps" (sinon le nouveau son coupait la musique). La musique est libre de droit. Conformément aux demandes du propriétaire de la musique, une mention vers la licence a été ajoutée dans le repo github et dans un fichier séparé.

FrancoisDuchene committed:
nostalgie

FrancoisDuchene committed:
upgrade des avions

FrancoisDuchene committed:
on commence un truc avec un chat (non terminé)

FrancoisDuchene committed:
RENOMMAGE

FrancoisDuchene committed:
sad_cat_face.mp3

Jimvy commented on app/src/main/java/lufra/youpidapp/Discotheque.kt:
```diff
@@ -79,16 +89,24 @@ class Discotheque(private val context: Context) {
         return mp
     }
 
+    // par défault on utilise "all" et pas "musics"
     private fun playOne(name: String): Int {
+        return playOne(name, all)
+    }
+
+    /**
+     * Joue le son "name" en mode "un seul à la fois". soundList est soit "all" soit "musics"
+     */
+    private fun playOne(name: String, soundList: HashMap<String, Int>): Int {
```
```suggestion
    /**
     * Joue le son "name" en mode "un seul à la fois".
     * @param name  le nom du son à jouer
     * @param soundList  une Map de nom de son (valides) vers un id de resource de son (valide).
     */
    private fun playOne(name: String, soundList: Map<String, Int> = all): Int {
```
En vrai il faudrait peut-être refactor ça et juste passer l'entier `soundList[name]!!`...

FrancoisDuchene replied:
En effet ça semble plus judicieux (j'avais oublié qu'en Kotlin on pouvait mettre des paramètres par défaut 😅 ). je vais refractor pour directement passer les id, ça semble plus logique que passer un hashmap

Jimvy commented on app/src/main/java/data/PaperPlane.kt:
```diff
@@ -0,0 +1,18 @@
+
+import android.view.View
+
+class PaperPlane(
```
Nitpicking : cette classe devrait plutôt se situer dans `BoiteFragment.kt` en tant qu'inner class vu qu'elle n'est utilisée que là, et parce qu'elle contient une View, donc ce n'est pas vraiment du data, mais du View (ou du ViewHolder ^^).

FrancoisDuchene replied:
J'ai bougé la classe hors de *BoiteFragment* parce que je trouvais que ça faisait déjà désordre dedans. Mais c'est vrai qu'elle est suffisamment petite pour que ça soit inutile de la mettre à part. (par contre je ne m'empêche pas de la redéplacer en dehors de *Boitefragment* dans le future pour réutiliser les avions autre part)

Jimvy replied:
On pourrait la mettre dans un package View ou un truc du style alors...
Ou alors peut-être que c'est à BoiteFragment d'externaliser une partie de son code pour la simplifier (je sais pas trop comment rendre ça clean note)

Jimvy commented on app/src/main/java/fragments/BoiteFragment.kt:
```diff
@@ -46,14 +56,123 @@ class BoiteFragment: MyFragment() {
+        comuBtn.setOnLongClickListener {
+            playSadCatAnimation()
+            context.discotheque.playMusiqueTriste()
+            context.discotheque.setType(2)
```
On devrait avoir une constante "PLAY_SEQUENTIAL" pour que ce soit plus clair ^^

TGLuis replied:
C'est vrai mais ça n'a rien à voir avec cette PR :p

FrancoisDuchene replied:
Je vais ajouter une issue là-dessus (pour pas oublier)

Jimvy reviewed (commented):
Faut que tu rebases/merges.

Comme suggéré dans un commentaire, peut-être qu'il faudrait refactor les méthodes pour qu'on puisse passer directement l'id de resources, plutôt que passer un (pointeur vers) string qui doit ensuite être passé dans une Map. Mais later...

La musique est un peu longue, et comme on sait, les blagues les plus courtes sont les meilleures ;)
Aussi, j'ai pas bien compris, mais pourquoi on passe le mode de lecture en "l'un après l'autre", et pas en "en même temps" ?

TGLuis commented on app/src/main/java/fragments/BoiteFragment.kt:
```diff
@@ -1,16 +1,24 @@
 package fragments
 
+import android.animation.*
```
Optimisation des imports? :p

FrancoisDuchene replied:
arf merci android studio :P

TGLuis commented on app/src/main/java/fragments/BoiteFragment.kt:
```diff
@@ -1,16 +1,24 @@
+import data.PaperPlane
 import lufra.youpidapp.MainActivity
 import lufra.youpidapp.R
+import java.util.*
```
Ici aussi, et d'ailleurs si le java util est importé juste pour le random, je pense que Kotlin en a un et que c'est mieux de pas utiliser les librairies java si on fait du kotlin.
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.random/-random/

FrancoisDuchene replied:
Ah ? J'ai réutilisé le code de Discothèque, il utilise aussi le random de java, je vais changer là aussi du coup

TGLuis commented on app/src/main/java/lufra/youpidapp/Discotheque.kt:
```diff
@@ -3,6 +3,7 @@ package lufra.youpidapp
+import android.util.Log
 import android.widget.Toast
 import java.lang.reflect.Field
 import java.util.*
```
Tant qu'on y est peut-être aussi refactor ici :p (je sais que c'est pas de cette pr mais bon :p )

FrancoisDuchene replied:
Attends j'ai pas compris, tu veux refractor quoi ici ?

Jimvy replied:
Retirer le Log ?

TGLuis reviewed (changes requested):
Plusieurs bonnes idées, qques corrections légères à faire d'après moi. Et je pense que le commentaire de JM sur la longueur de la musique est pertinent :+1:

FrancoisDuchene commented:
> Faut que tu rebases/merges.

j'ai peur de faire de la merde (je maitrise pas vraiment les rebase)

> Comme suggéré dans un commentaire, peut-être qu'il faudrait refactor les méthodes pour qu'on puisse passer directement l'id de resources, plutôt que passer un (pointeur vers) string qui doit ensuite être passé dans une Map. Mais later...

Je vais changer uniquement pour la méthode qui nous intéresse ici (```playOne```), à voir plus tard si il faut refractor d'autres parties

> La musique est un peu longue, et comme on sait, les blagues les plus courtes sont les meilleures ;)

C'était un peu la blague aussi, mais je vais diminuer à 30 secondes (là c'est 60)

> Aussi, j'ai pas bien compris, mais pourquoi on passe le mode de lecture en "l'un après l'autre", et pas en "en même temps" ?

10 secondes avant la fin de la musique, y a un "une caste inférieure" qui est lancé. Si c'est dans un autre mode que "en même temps", la musique est coupée pour lancer le nouveau son. Or je veux qu'il soit lancé pendant que la musique tourne. Maintenant je viens de me rendre compte que je reset jamais vers l'ancien mode

FrancoisDuchene committed:
the cat is a lie

fait les modifs suggérées dans #25

Jimvy commented:
> > Faut que tu rebases/merges.
> 
> j'ai peur de faire de la merde (je maitrise pas vraiment les rebase)

No problem, merge alors
> > Comme suggéré dans un commentaire, peut-être qu'il faudrait refactor les méthodes pour qu'on puisse passer directement l'id de resources, plutôt que passer un (pointeur vers) string qui doit ensuite être passé dans une Map. Mais later...
>
> Je vais changer uniquement pour la méthode qui nous intéresse ici (`playOne`), à voir plus tard si il faut refractor d'autres parties

👍
> > La musique est un peu longue, et comme on sait, les blagues les plus courtes sont les meilleures ;)
>
> C'était un peu la blague aussi, mais je vais diminuer à 30 secondes (là c'est 60)

Ou même 15-20, mais bon.

> > Aussi, j'ai pas bien compris, mais pourquoi on passe le mode de lecture en "l'un après l'autre", et pas en "en même temps" ?
>
> 10 secondes avant la fin de la musique, y a un "une caste inférieure" qui est lancé. Si c'est dans un autre mode que "en même temps", la musique est coupée pour lancer le nouveau son. Or je veux qu'il soit lancé pendant que la musique tourne. Maintenant je viens de me rendre compte que je reset jamais vers l'ancien mode

Oui oui, ma question c'était pourquoi "one after one" et pas "en même temps" : ici ça marche car "one after one" ne coupe pas le son en cours (cf #24 ), mais en vrai "en même temps" serait plus accurate.

FrancoisDuchene committed:
pluions nous aux exigences

finalement j'ai (un peu) refractor Discotheque.kt pour utiliser les id de son
j'ai ajouté un son de pluie pour aller avec le chat

FrancoisDuchene commented:
> No problem, merge alors

je fais dès que Luis voie les changements 😉 
 
> > > Comme suggéré dans un commentaire, peut-être qu'il faudrait refactor les méthodes pour qu'on puisse passer directement l'id de resources, plutôt que passer un (pointeur vers) string qui doit ensuite être passé dans une Map. Mais later...
> > 
> > 
> > Je vais changer uniquement pour la méthode qui nous intéresse ici (`playOne`), à voir plus tard si il faut refractor d'autres parties

> Oui oui, ma question c'était pourquoi "one after one" et pas "en même temps" : ici ça marche car "one after one" ne coupe pas le son en cours (cf #24 ), mais en vrai "en même temps" serait plus accurate.

bon ben finalement j'ai fait pour playStack et playList aussi. J'ai ajouté un son de pluie qui joue en même temps que la musique. Au lieu d'appeler ```playOne``` pour jouer la musique, j’appelle ```playStack```.

TGLuis reviewed (approved):
Ok for me :)
